tests/Unit/CSSList/DocumentTest.php

```diff
@@ -142,4 +149,92 @@ public function insertContentBeforeAppendsIfSiblingNotFound(): void
         self::assertCount(4, $subject->getContents());
         self::assertSame([$bogusOne, $sibling, $bogusTwo, $item], $subject->getContents());
     }
+
+    /**
+     * @test
+     */
+    public function getAllDeclarationBlocksForNoContentsReturnsEmptyArray(): void
+    {
+        $subject = new Document();
+
+        self::assertSame([], $subject->getAllDeclarationBlocks());
+    }
+
+    /**
+     * @test
+     */
+    public function getAllDeclarationBlocksCanReturnOneDirectDeclarationBlockContent(): void
+    {
+        $subject = new Document();
+
+        $declarationBlock = new DeclarationBlock();
+        $subject->setContents([$declarationBlock]);
+
+        $result = $subject->getAllDeclarationBlocks();
+
+        self::assertSame([$declarationBlock], $result);
+    }
+
+    /**
+     * @test
+     */
+    public function getAllDeclarationBlocksCanReturnMultipleDirectDeclarationBlockContents(): void
+    {
+        $subject = new Document();
+
+        $declarationBlock1 = new DeclarationBlock();
+        $declarationBlock2 = new DeclarationBlock();
+        $subject->setContents([$declarationBlock1, $declarationBlock2]);
+
+        $result = $subject->getAllDeclarationBlocks();
+
+        self::assertSame([$declarationBlock1, $declarationBlock2], $result);
+    }
+
+    /**
+     * @test
+     */
+    public function getAllDeclarationBlocksAlsoReturnsDeclarationBlocksWithinAtRuleBlockList(): void
+    {
+        $subject = new Document();
+
+        $declarationBlock = new DeclarationBlock();
+        $atRuleBlockList = new AtRuleBlockList('media');
+        $atRuleBlockList->setContents([$declarationBlock]);
+        $subject->setContents([$atRuleBlockList]);
+
+        $result = $subject->getAllDeclarationBlocks();
+
+        self::assertSame([$declarationBlock], $result);
+    }
+
+    /**
+     * @test
+     */
+    public function getAllDeclarationBlocksIgnoresImport(): void
+    {
+        $subject = new Document();
+
+        $import = new Import(new URL(new CSSString('')), '');
```

Comment:
A nonempty and valid URL like `https://example.org/` would be better here.

Reply:
Done.